src/application/snapshot/widgets/organization-proposals.container.tsx

```diff
@@ -0,0 +1,87 @@
+import { useState } from 'react';
+
+import { useCommandQuery } from 'shared/messaging';
+import { Pagination } from 'shared/ui';
+
+import { GetProposalCommand } from '../commands';
+
+import { Proposal } from './proposal';
+
+interface Properties {
+  snapshotHandle: string;
+  className?: string;
+  top?: number;
+  onClose?: () => void;
+}
+
+export const OrganizationProposalsContainer = ({
+  snapshotHandle,
+  className = 'fixed top-20',
+  top,
+  onClose,
+}: Properties) => {
+  const [currentProposalIndex, setCurrentProposalIndex] = useState(0);
+
+  const proposalQuery = useCommandQuery({
+    command: new GetProposalCommand({
+      snapshotName: snapshotHandle ?? '',
+      pageNumber: currentProposalIndex,
+    }),
+    enabled: snapshotHandle ? snapshotHandle.length > 0 : false,
+    staleTime: Number.POSITIVE_INFINITY,
+    placeholderData: (previousData) => {
+      return previousData;
+    },
+  });
+
+  const currentProposal = proposalQuery.data?.proposal;
+
+  const isLoadingProposal =
+    proposalQuery.isLoading || proposalQuery.isPlaceholderData;
+
+  const isPreviousProposalAvailable = currentProposalIndex > 0;
+  const isNextProposalAvailable =
+    Boolean(proposalQuery.data?.hasNextProposal) && !isLoadingProposal;
+
+  const showPreviousProposal = () => {
+    if (!isPreviousProposalAvailable || isLoadingProposal) {
+      return;
+    }
+
+    setCurrentProposalIndex((previous) => {
+      return previous - 1;
+    });
+  };
```

Comment:
maybe we should disable previousProposal while isLoading - same as you did for isNextProposalAvailable and later inside `showPreviousProposal` you can only check if it's available without need to check for loading state

Reply:
oh yeah, but as I'm thinking about it, checking isLoading as a condition to show previous proposals doesn't make sense as we cache them. I've removed the isLoadingProposal from that condition. Let me know what you think about that.

Reply:
makes sense, thanks 👍🏻